core_backend/app/users/models.py

```diff
@@ -0,0 +1,88 @@
+from datetime import datetime
+from typing import Optional
+
+from sqlalchemy import (
+    DateTime,
+    String,
+    # delete,
+    select,
+)
+from sqlalchemy.exc import NoResultFound
+from sqlalchemy.ext.asyncio import AsyncSession
+from sqlalchemy.orm import Mapped, mapped_column
+
+from ..models import Base
+from .schemas import UserCreate
+
+
+class UserDB(Base):
+    """
+    SQL Alchemy data model for users
+    """
+
+    __tablename__ = "user"
+
+    user_id: Mapped[str] = mapped_column(String, primary_key=True, nullable=False)
+    username: Mapped[str] = mapped_column(String, nullable=False)
```

Comment:
What is the value in having a userid and a username? Looks like we are allowing username to be not unique and userid to unique. But later we retrieve userid by username.

Reply:
Good point - in the application logic I'm assuming username is unique for now. I'm anticipating that when we offload the user management to a OAuth service, it will just send back a unique ID per user (which would replace username) and thought we might want to keep our own generated user_id?

Also, we can control and standardise user_ids but usernames can be anything... It's definitely best practice to keep a generated ID but I get that right now we essentially have 2 equivalent ID columns...

Not sure - maybe I could just use username as ID for now. Or if you're okay with it, we keep this until we get to find out how OAuth really works and adjust there.